src/rust/ensogl/lib/theme/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+//! Application theme setup.
+
+/// `define_theme` helper.
+macro_rules! _define_theme_literals {
+    ([$theme_name:ident $($path:ident)*] $name:ident = $e:expr) => {
+        $theme_name.insert(stringify!($($path.)*$name), $e);
+    };
+
+    ([$($path:ident)*] $name:ident = $e:expr; $($rest:tt)*) => {
+        _define_theme_literals!([$($path)*] $name = $e);
+        _define_theme_literals!([$($path)*] $($rest)*);
+    };
+
+    ([$($path:ident)*] $name:ident {$($t:tt)*}) => {
+        _define_theme_literals!([$($path)* $name] $($t)*);
+    };
+
+    ([$($path:ident)*] $name:ident {$($t:tt)*} $($rest:tt)*) => {
+        _define_theme_literals!([$($path)*] $name {$($t)*});
+        _define_theme_literals!([$($path)*] $($rest)*);
+    };
+}
+
+macro_rules! _define_theme_modules {
+    ([$theme_name:ident $($path:ident)*] $name:ident = $e:expr) => {
+        pub const $name : &str = stringify!($($path.)*$name);
+    };
+
+    ([$($path:ident)*] $name:ident = $e:expr; $($rest:tt)*) => {
+        _define_theme_modules!([$($path)*] $name = $e);
+        _define_theme_modules!([$($path)*] $($rest)*);
+    };
+
+    ([$($path:ident)*] $name:ident {$($t:tt)*}) => {
+        pub mod $name {
+            _define_theme_modules!([$($path)* $name] $($t)*);
+        }
+    };
+
+    ([$($path:ident)*] $name:ident {$($t:tt)*} $($rest:tt)*) => {
+        _define_theme_modules!([$($path)*] $name {$($t)*});
+        _define_theme_modules!([$($path)*] $($rest)*);
+    };
+}
+
+/// Used to define default theme.
+#[macro_export]
+macro_rules! define_default_theme {
+    ($name:ident $($t:tt)*) => {
+        define_theme!($name $($t)*);
+
+        #[allow(non_upper_case_globals)]
+        #[allow(missing_docs)]
+        pub mod vars {
+            _define_theme_modules!([$name] $($t)*);
+        }
+    };
+}
+
+/// Used to define any theme.
+#[macro_export]
+macro_rules! define_theme {
+    ($name:ident $($t:tt)*) => {
+        #[allow(missing_docs)]
+        pub mod $name {
+            use ensogl_core::application::Application;
+            use ensogl_core::data::color;
+            use ensogl_core::display::style::theme;
+
+            /// Setup the $name theme in application.
+            pub fn setup(app:&Application) {
+                let mut $name = theme::Theme::new();
+                _define_theme_literals!([$name] $($t)*);
+                app.themes.register(stringify!($name),$name);
+                app.themes.set_enabled(&[stringify!($name)]);
+            }
+        }
+    };
+}
+
+define_theme! { dark
+    application {
+        background {
+            color = color::Lcha::new(0.13,0.013,0.18,1.0)
+        }
+        text {
+            color = color::Lcha::new(1.0,0.0,0.0,0.7);
+            selection {
+                color = color::Lcha::new(0.7,0.0,0.125,0.7)
+            }
+        }
```

Comment:
We have the same variable for text over nodes and in TextEditor. I would rather make separately `graph_editor.node.text.{color/selection}` and `text_editor.text.{color/selection}`. One can then just set the color globally by setting `text.{color/selection}` (this is how cascade styles work) in some non-default theme.